internal/server/server.go

```diff
@@ -100,8 +102,11 @@ func (s *Server) LogIn(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	// hash.ComparePasswords(request.Password, user.Password) -> next PR with hashing
-
+	if ok, err := hash.ComparePasswords(user.Password, request.Password); !ok || err != nil {
+		http.Error(w, "invalid email or password: "+err.Error(), http.StatusUnauthorized)
+		return
+	}
```

Comment:
We don't want to give away any information about why password and email are not valid

Reply:
> 
> 
> We don't want to give away any information about why password and email are not valid

Ok, this information is needed only as a result of registration error